tests/chez/chez003/IORef.idr

```diff
@@ -14,3 +14,14 @@ main
          printLn val
          val <- readIORef y
          printLn val
+         z   <- newIORef 42
+         val <- readIORef z
+         printLn val
+         t1 <- fork $ do
+           atomicModifyIORef z (* 2)
+         t2 <- fork $
+           atomicModifyIORef z (* 2)
```

Comment:
In order to proof atomicity, the two competing threads must be long running (hundreds of milliseconds) and both access the mutable reference *a lot* (thousands or better millions of accesses). Preferably, one is considerably slower than the other. Only then will you create enough contention to truly show atomicity. Again, see my example on discord.

In this example, creating and acquiring the mutex (which you currently do on every call to `atomicModifyIORef`) is so slow (mutexes are slow) compared to the simple update action, that you will almost always get the illusion of atomicity. Try running your version of `atomicModifyIORef` with the example code I gave on discord and you will see that it is not atomic at all.

Reply:
Ah I see, the code I added in `chez003` never actually created any true contention thus never tested the atomicity of `atomicModifyIORef`.  Thanks for helping me see/understand this!